src/common/libsubprocess/subprocess.h

```diff
@@ -61,6 +61,13 @@ enum {
     FLUX_SUBPROCESS_FLAGS_SETPGRP = 2,
     /* use fork(2)/exec(2) even if posix_spawn(3) available */
     FLUX_SUBPROCESS_FLAGS_FORK_EXEC = 4,
+    /* flux_rexec(): In order to improve performance, do not locally
+     * copy and buffer output from the remote subprocess.  Immediately
+     * call output callbacks.  Users can directly retrieve data via
+     * flux_subprocess_read().  Data will not NUL terminated.  Use of
+     * other read functions will result in a EPERM error.
+     */
+    FLUX_SUBPROCESS_FLAGS_LOCAL_UNBUF = 8,
```

Comment:
>   Users can directly retrieve data via `flux_subprocess_read()`.

We probably should specify that `flux_subprocess_read()` must be called exactly once when the output callback is invoked.  If not called, data is lost.  If called more than once, the same data is returned.

> Data will not NUL terminated.

Missing a "be".

> Use of other read functions will result in a EPERM error.

EPERM is kind of strange here.  Would EINVAL be better?

Reply:
> EPERM is kind of strange here. Would EINVAL be better?

I currently use EPERM for another case where it is "illegal" to call a certain function.  It seems an ok choice.  To me EINVAL just makes you think bad input.  Hmmm, a better option isn't coming to me.  Will sleep on it.

Reply:
No big deal I suppose!